pkg/alertmanager/multitenant.go

```diff
@@ -405,15 +402,21 @@ func (am *MultitenantAlertmanager) setConfig(cfg alerts.AlertConfigDesc) error {
 	} else {
 		userAmConfig, err = amconfig.Load(cfg.RawConfig)
 		if err != nil && hasExisting {
-			// XXX: This means that if a user has a working configuration and
-			// they submit a broken one, we'll keep processing the last known
-			// working configuration, and they'll never know.
-			// TODO: Provide a way of communicating this to the user and for removing
-			// Alertmanager instances.
+			// This means that if a user has a working config and
+			// they submit a broken one, the Manager will keep running the last known
+			// working configuration.
 			return fmt.Errorf("invalid Cortex configuration for %v: %v", cfg.User, err)
 		}
 	}
 
+	// We can have an empty configuration here if:
+	// 1) the user had a previous alertmanager
+	// 2) then, submitted a non-working configuration (and we kept running the prev working config)
+	// 3) finally, the cortex AM instance is restarted and the running version is no longer present
+	if userAmConfig == nil {
```

Comment:
**suggestion(non-blocking,if-minor):** Since this check has been moved from `transformConfig` consider removing the `transformConfig` function entirely an moving the for loop logic from the function here. `transformConfig` is only called here so it should be safe to move and I think it would still be quite readable as part of a single function call.